static/app/components/workflowEngine/form/control/priorityControl.tsx

```diff
@@ -0,0 +1,189 @@
+import {useCallback, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {GroupPriorityBadge} from 'sentry/components/badge/groupPriority';
+import {Chevron} from 'sentry/components/chevron';
+import {CompactSelect, type SelectOption} from 'sentry/components/compactSelect';
+import {Flex} from 'sentry/components/container/flex';
+import FieldWrapper from 'sentry/components/forms/fieldGroup/fieldWrapper';
+import NumberField from 'sentry/components/forms/fields/numberField';
+import InteractionStateLayer from 'sentry/components/interactionStateLayer';
+import {IconArrow} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import {PriorityLevel} from 'sentry/types/group';
+
+export interface PriorityControlGridProps {
+  name: string;
+  /** Number of rows to display (defaults to 3) */
+  limit?: number;
+}
+
+function getPriority(item: number, limit: number): PriorityLevel {
+  if (item === limit - 1) {
+    return PriorityLevel.HIGH;
+  }
+  if (limit === 3 && item === 0) {
+    return PriorityLevel.LOW;
+  }
+  return PriorityLevel.MEDIUM;
+}
+
+/**
+ * TODO(nate): expose as controlled component that accepts onChange
```

Comment:
👍 sg - are you thinking the result data structure would be: `Record<PriorityLevel, number>`?

Reply:
https://github.com/getsentry/sentry/blob/86b96723732f1848893d700191e1311c4f978e40/static/app/components/workflowEngine/form/control/priorityControl.tsx#L22-L26

Reply:
👍 we can discuss this shape in that pr for the future :) (i think there might be a typo on the `created` bit :P)